api/agent/uniter/unit_test.go

```diff
@@ -975,3 +1411,16 @@ func (s *unitSuite) TestCanApplyLXDProfile(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(canApply, jc.IsTrue)
 }
+
+func (s *unitSuite) TestCanApplyLXDProfileNotImplemented(c *gc.C) {
+	apiCaller := basetesting.APICallerFunc(func(objType string, version int, id, request string, arg, result interface{}) error {
+		return apiservererrors.ServerError(errors.NotImplementedf("not implemented"))
+	})
+	tag := names.NewUnitTag("mysql/0")
+	client := uniter.NewState(apiCaller, tag)
+
+	unit := uniter.CreateUnit(client, names.NewUnitTag("mysql/0"))
+
+	_, err := unit.CanApplyLXDProfile()
+	c.Assert(err, jc.ErrorIs, errors.NotImplemented)
+}
```

Comment:
I don't suppose there's anything you can do to avoid the amount of duplication here? This gets a little silly!

Perhaps a sub-suite, who's setup steps configure the unit?

Reply:
Yeah, I'm loathed to re-write the hole of this test suite. If I see it again, I will.

Good shout though.